src/app/helptext/task-calendar/snapshot/snapshot-form.ts

```diff
@@ -1,49 +1,39 @@
 import { T } from '../../../translate-marker';
-import {Validators} from '@angular/forms';
+
 
 export default {
-    task_filesystem_placeholder: T('Pool/Dataset'),
-    task_filesystem_tooltip: T('Select a pool, dataset, or zvol.'),
-    task_filesystem_validation : [ Validators.required ],
-
-    task_recursive_placeholder: T('Recursive'),
-    task_recursive_tooltip: T('Set this to take separate snapshots of the\
-                pool/dataset and each of its child datasets. Leave\
-                unset to take a single snapshot of the specified\
-                pool/dataset with <b>no</b> child datasets.'),
-
-    task_ret_count_placeholder: T('Snapshot Lifetime'),
-    task_ret_count_validation: [Validators.min(0)],
-
-    task_ret_unit_tooltip: T('Define a length of time to retain the snapshot on this\
-                system. After the time expires, the snapshot is removed.\
-                Snapshots which have been replicated to other systems\
-                are not affected.'),
-
-                type: 'select',
-    name: 'task_begin',
-    task_begin_placeholder: T('Begin'),
-    task_begin_tooltip: T('Choose the hour and minute when the system can begin\
-                taking snapshots.'),
-    options: [],
-    value: '',
-    required: true,
-    task_begin_validation : [ Validators.required ],
-
-    task_end_placeholder: T('End'),
-    task_end_tooltip: T('Choose the hour and minute when the system must stop\
-                taking snapshots.'),
-    task_end_validation : [ Validators.required ],
-
-    task_interval_placeholder: T('Interval'),
-    task_interval_tooltip: T('Define how often the system takes snapshots between the\
-                <b>Begin</b> and <b>End</b> times.'),
-    task_interval_validation : [ Validators.required ],
-
-    task_byweekday_placeholder: T('Day of week'),
-    task_byweekday_tooltip: T('Choose the days of the week to take snapshots.'),
-    task_byweekday_validation : [ Validators.required ],
-
-    task_enabled_placeholder: T('Enabled'),
-    task_enabled_tooltip: T('Unset to disable this task without deleting it.'),
+    dataset_placeholder: T('Pool/Dataset'),
+    dataset_tooltip: T('Select a pool, dataset, or zvol.'),
+
+    recursive_placeholder: T('Recursive'),
+    recursive_tooltip: T('Set this to take separate snapshots of the\
+ pool/dataset and each of its child datasets. Leave unset to take a\
+ single snapshot of the specified pool/dataset with <b>no</b> child datasets.'),
+
+    exclude_placeholder: T('Exclude'),
+    exclude_tooltip: T(''),
+
+    lifetime_value_placeholder: T('Snapshot Lifetime'),
+
+    lifetime_unit_tooltip: T('Define a length of time to retain the snapshot on this\
+ system. After the time expires, the snapshot is removed. Snapshots which have\
+ been replicated to other systems are not affected.'),
+
+    naming_schema_placeholer: T('Naming Schema'),
+    naming_schema_tooltip: T(''),
```

Comment:
Rough long form suggestion from FreeNAS API docs and https://github.com/freenas/zettarepl:

```
Define how snapshots are named, based on creation date. Avoid creating similar naming schema. Schema must include these format strings: <i>"%Y", "%m","%d", "%H" and "%M"</i>. Example schema: <i>snaptask-%Y-%d-%m-%H-%M</i>.

Snapshot names include the item being snapshot and the snapshot lifetime. Example name: <i>pool1@snaptask-20190315.0527-2w</i>. This is an automated snapshot of <i>pool1</i>, created on March 15, 2019 at 5:27am. It has a lifetime of two weeks.
```

Reply:
"Snapshot name format string. The default is *snap-%Y-%m-%d-%H-%M*.
Must include the strings %Y, %m, %d, %H, and %M, which are replaced with the four-digit year, month, day of month, hour, and minute as defined in strftime(3). A string showing the snapshot lifetime is appended to the name. For example, snapshots of pool1 with a Naming Schema of "customsnap-%Y%m%d.%H%M" and lifetime of two weeks have names like *pool1@customsnap-20190315.0527-2w*."

Reply:
Still investigating, need to find what the default naming schema (format) is.

Reply:
the default value for naming schema is `auto-%Y-%m-%d_%H-%M`

Reply:
@william-gr  what are your thoughts?

Reply:
LGTM